contracts/InterchainTokenService.sol

```diff
@@ -476,20 +478,13 @@ contract InterchainTokenService is
         string calldata destinationChain,
         bytes calldata destinationAddress,
         uint256 amount,
-        bytes calldata data
+        bytes memory data
     ) external payable whenNotPaused {
         ITokenManager tokenManager = ITokenManager(tokenManagerAddress(tokenId));
 
         amount = tokenManager.takeToken(msg.sender, amount);
 
-        _transmitInterchainTransfer(
-            tokenId,
-            msg.sender,
-            destinationChain,
-            destinationAddress,
-            amount,
-            abi.encodePacked(LATEST_METADATA_VERSION, data)
-        );
+        _transmitInterchainTransfer(tokenId, msg.sender, destinationChain, destinationAddress, amount, LATEST_METADATA_VERSION, data);
```

Comment:
Not sure when LATEST_METADATA_VERSION became a constant, but here we should use a more specific constant )which is the same as LATEST_METADATA_VERSION for now) which would be named something like LATEST_TRANSMIT_INTERCHAIN_TRANSFER_METADATA_VERSION.